File: benchmark/output/lua-love/openai/heat_diffusion.lua
local heat = {}
local next_heat = {}

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("2D Heat Diffusion")
    for i = 1, 192 do
        table.insert(heat, 0.0)
        table.insert(next_heat, 0.0)
    end
end

function love.mousepressed(mx, my, button)
    local col = math.floor(mx / 50)
    local row = math.floor(my / 50)
    if col >= 0 and col < 16 and row >= 0 and row < 12 then
        heat[row * 16 + col + 1] = 100.0
    end
end

function love.update(dt)
    for row = 0, 11 do
        for col = 0, 15 do
            local idx = row * 16 + col + 1
            local sum = 0.0
            local n = 0
            if row > 0 then
                sum = sum + heat[(row - 1) * 16 + col + 1]
                n = n + 1
            end
            if row < 11 then
                sum = sum + heat[(row + 1) * 16 + col + 1]
                n = n + 1
            end
            if col > 0 then
                sum = sum + heat[row * 16 + col]
                n = n + 1
            end
            if col < 15 then
                sum = sum + heat[row * 16 + col + 2]
                n = n + 1
            end
            local avg = sum / n
            next_heat[idx] = heat[idx] + (avg - heat[idx]) * 2.0 * dt
            next_heat[idx] = next_heat[idx] * 0.998
        end
    end
    for i = 1, 192 do
        heat[i] = next_heat[i]
    end
end

function love.draw()
    local max_t = 0.0
    for row = 0, 11 do
        for col = 0, 15 do
            local temp = heat[row * 16 + col + 1]
            if temp > max_t then
                max_t = temp
            end
            local size = temp / 100.0 * 48.0
            if size > 48.0 then
                size = 48.0
            end
            if size > 1.0 then
                local off = (48.0 - size) / 2.0
                love.graphics.rectangle("fill", col * 50 + 1 + off, row * 50 + 1 + off, size, size)
            end
        end
    end
    love.graphics.setColor(1, 1, 1)
    love.graphics.print("Click to add heat", 10, 10)
    love.graphics.print("Max: " .. math.floor(max_t), 10, 30)
end

File: vibe-lang/benchmark/output/python-pygame/openai/heat_diffusion.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("2D Heat Diffusion")
clock = pygame.time.Clock()

heat = [0.0] * 192
next_heat = [0.0] * 192

def mousepressed(mx, my, button):
    col = int(mx / 50.0)
    row = int(my / 50.0)
    if 0 <= col < 16 and 0 <= row < 12:
        heat[row * 16 + col] = 100.0

def update(dt):
    for row in range(12):
        for col in range(16):
            idx = row * 16 + col
            sum_heat = 0.0
            n = 0
            if row > 0:
                sum_heat += heat[(row - 1) * 16 + col]
                n += 1
            if row < 11:
                sum_heat += heat[(row + 1) * 16 + col]
                n += 1
            if col > 0:
                sum_heat += heat[row * 16 + col - 1]
                n += 1
            if col < 15:
                sum_heat += heat[row * 16 + col + 1]
                n += 1
            avg = sum_heat / float(n)
            next_heat[idx] = heat[idx] + (avg - heat[idx]) * 2.0 * dt
            next_heat[idx] *= 0.998
    for i in range(192):
        heat[i] = next_heat[i]

def draw():
    screen.fill((0, 0, 0))
    max_t = 0.0
    for row in range(12):
        for col in range(16):
            temp = heat[row * 16 + col]
            if temp > max_t:
                max_t = temp
            size = temp / 100.0 * 48.0
            if size > 48.0:
                size = 48.0
            if size > 1.0:
                off = (48.0 - size) / 2.0
                pygame.draw.rect(screen, (255, 255, 255), (col * 50.0 + 1.0 + off, row * 50.0 + 1.0 + off, size, size))
    font = pygame.font.Font(None, 36)
    text = font.render("Click to add heat", True, (255, 255, 255))
    screen.blit(text, (10, 10))
    text = font.render("Max: " + str(int(max_t)), True, (255, 255, 255))
    screen.blit(text, (10, 30))

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.MOUSEBUTTONDOWN:
            mx, my = pygame.mouse.get_pos()
            mousepressed(mx, my, event.button)

    update(dt)
    draw()
    pygame.display.flip()

pygame.quit()
sys.exit()